doc/dev/custom_devices.rst

```diff
@@ -17,91 +17,116 @@ Additionally, all measurements will always return ``true``.
-            virtual void Counts(DataView<double, 1> &, DataView<int64_t, 1> &, size_t) {}
-
-            virtual void PartialCounts(DataView<double, 1> &, DataView<int64_t, 1> &,
-                                       const std::vector<QubitIdType> &, size_t)
+            auto Expval(ObsIdType) -> double override { return 0.0; }
+            auto Var(ObsIdType) -> double override { return 0.0; }
+            void State(DataView<std::complex<double>, 1> &) override {}
+            void Probs(DataView<double, 1> &) override {}
+            void PartialProbs(DataView<double, 1> &, const std::vector<QubitIdType> &) override {}
+            void Sample(DataView<double, 2> &, size_t) override {}
+            void PartialSample(DataView<double, 2> &, const std::vector<QubitIdType> &, size_t) override {}
+            void Counts(DataView<double, 1> &, DataView<int64_t, 1> &, size_t) override {}
+
+            void PartialCounts(DataView<double, 1> &, DataView<int64_t, 1> &,
+                                    const std::vector<QubitIdType> &, size_t) override
             {
             }
 
-            virtual void Gradient(std::vector<DataView<double, 1>> &, const std::vector<size_t> &) {}
+            void Gradient(std::vector<DataView<double, 1>> &, const std::vector<size_t> &) override {}
         };
 
-In addition to implementing the ``QuantumDevice`` class, one must implement the following method:
+In addition to implementing the ``QuantumDevice`` class, one must implement the following method
+with the name ``CustomDevice`` + ``Factory``:
 
 .. code-block:: c++
 
     extern "C" Catalyst::Runtime::QuantumDevice*
-    getCustomDevice() { return new CustomDevice(); }
+    CustomDeviceFactory(bool status, const std::string &kwargs)
+    {
+        return new CustomDevice(status, kwargs); }
+
+The use of the term ``Factory`` in ``CustomDeviceFactory`` is a runtime convention that combines the name
+of the device class, in this case, ``CustomDevice``, with the suffix ``Factory``. This signifies that the
+primary purpose of the method is to act as a factory for creating instances of the ``CustomDevice`` class.
+
+.. warning::
+
+    The runtime exclusively supports factory methods with the name of the device class followed by ``Factory``.
+    Please ensure that your method adheres to this naming convention.
 
-where ``CustomDevice()`` is a constructor for your custom device.
-``CustomDevice``'s destructor will be called by the runtime.
+``CustomDevice(status, kwargs)`` serves as a constructor for your custom device, with
 
-.. note::
+- ``status``: A boolean to control the initial value of the tape recording system in the runtime.
+  If ``true``, this will cache the entire operations and observables in a program.
+  This may be used to compute gradient of a quantum circuit with taking advantage of gradient
+  methods provided by backend devices if supported.
```

Comment:
You mentioned that the tape recording system can be used by any device, but here it looks like the device still needs to keep track of the recording state?

Reply:
I'm wondering if we need to either explain this component more, since the plugin developer is required to accept this parameter, or if we'd like to remove it somehow?

Reply:
Giving it some thoughts and this is not required to be part of the constructor anymore as the `StartTapeRecording` and `StopTapeRecording` methods in the QuantumDevice interface properly satisfies the  objectives of this `status` arg in the constructor. I removed this and updated the docs.